components/gfx/paint_context.rs

```diff
@@ -775,8 +774,7 @@ impl<'a> PaintContext<'a> {
 
         // You can't scale black color (i.e. 'scaled = 0 * scale', equals black).
         let mut scaled_color = color::black();
-        // TODO(Savago): Use equality operators when we sync with rust-azure.
-        if color.r != scaled_color.r || color.g != scaled_color.g || color.b != scaled_color.b {
```

Comment:
There is also a bug here where transparent black and black are considered the same color. I think this change fixes that.